snapcraft.yaml

```diff
@@ -19,44 +32,57 @@ slots:
 apps:
   evince:
     command: usr/bin/evince
-    extensions: [gnome-3-38]
+    extensions: [gnome]
     plugs:
       - avahi-observe
-      - cups-control
+      - cups
       - gsettings
       - home
       - network
       - removable-media
     desktop: usr/share/applications/org.gnome.Evince.desktop
+
   evince-previewer:
     command: usr/bin/evince-previewer
     plugs:
       - avahi-observe
-      - cups-control
+      - cups
       - gsettings
       - home
       - network
     desktop: usr/share/applications/org.gnome.Evince-previewer.desktop
+
   evince-thumbnailer:
     command: usr/bin/evince-thumbnailer
     plugs:
       - gsettings
       - home
 
 parts:
+  buildenv:
+    plugin: nil
+    build-environment: &buildenv
+      - PKG_CONFIG_PATH: $CRAFT_STAGE/usr/lib/$CRAFT_ARCH_TRIPLET/pkgconfig${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}
+
   evince:
     source: https://gitlab.gnome.org/GNOME/evince.git
     source-type: git
-    source-branch: 'gnome-41'
+    source-tag: '42.3'
     plugin: meson
+    parse-info: [usr/share/metainfo/org.gnome.Evince.appdata.xml]
+    build-environment: *buildenv
```

Comment:
since the evince part is the only one that uses the `PKG_CONFIG_PATH` set in the buildenv part, just define it here instead and remove that buildenv part.

Reply:
Good point. Done.